### PR TITLE
Project refactoring

### PR DESCRIPTION
**Pull request by regiskuckaertz**
There's a couple of systemic changes that I wanted to go through before improving the tooling:

- create a separate `atom-renderer-email` project to remove the dependency on jsoup in the core project
- move to SBT 1
- upgrade all Scala deps
- upgrade all JS deps
- improve the build script (heavily inspired by @hosamaly's work on [nitf-scala](https://github.com/guardian/nitf-scala))

There's a couple of PRs after this one to move the whole build pipeline into SBT, improve the documentation and add tests.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,77 +1,86 @@
...
-    ScmInfo(
-      url("https://github.com/guardian/atom-renderer"),
-      "scm:git@github.com:guardian/atom-renderer.git"
-    )
-  ),
-  developers := List(
-    Developer(id="regiskuckaertz", name="Regis Kuckaertz", email="regis.kuckaertz@theguardian.com", url=url("https://github.com/regiskuckaertz"))
+lazy val root = (project in file("."))
+  .aggregate(core, email, utils)
+  .settings(commonSettings, disablePublishingSettings)
```
Comment (hosamaly) [inline build.sbt]: I didn't know you could pass multiple sequences as parameters to this method. Nice!

File: build.sbt [hunk 2]
```diff
@@ -1,77 +1,86 @@
...
+  .settings(commonSettings, utilsSettings, disablePublishingSettings)
+
+lazy val commonSettings: Seq[Setting[_]] = Metadata.settings ++ 
+  Seq ( crossScalaVersions := scalaVersions
+      , scalacOptions ++= Seq("-feature", "-deprecation", "-target:jvm-1.8", "-language:higherKinds")
+      , scalaVersion := scalaVersions.min
+      , libraryDependencies ++= coreDeps
+      , dependencyOverrides += "org.apache.thrift" % "libthrift" % "0.9.1"
+      , unmanagedResourceDirectories in Compile += (baseDirectory in ThisBuild).value / "build"
+      , excludeFilter in Compile in unmanagedResources := "*.fjs" || "*.scss"
```
Comment (hosamaly) [inline build.sbt]: Starting with [sbt 1.1.0](https://github.com/sbt/sbt/releases/tag/v1.1.0), this line can also be written using the unified slash syntax:
```sbt
Compile / unmanagedResources / excludeFilter := "*.fjs" || "*.scss"
```
Whichever syntax we use, we should probably unify it across the file (e.g. `Compile / sources` vs `sources in Compile`).

File: build.sbt [hunk 3]
```diff
@@ -1,77 +1,86 @@
...
+        """import monix.execution.Scheduler.Implicits.global
+          |import com.gu.contentatom.thrift._
+          |import com.gu.contentatom.renderer._
+          |import com.gu.contentatom.renderer.utils._
+        """.stripMargin
+      )
+
+
+lazy val twirlSettings: Seq[Setting[_]] = 
+  Seq (sourceDirectories in (Compile, TwirlKeys.compileTemplates) += (resourceDirectory in Compile).value)
```
Comment (hosamaly) [inline build.sbt]: This is also another instance where we could use the new syntax.
```sbt
Compile / TwirlKeys.compileTemplates / sourceDirectories += (Compile / resourceDirectory).value
```

File: project/Dependencies.scala [hunk 4]
```diff
@@ -1,17 +1,19 @@
 import sbt._
 
 object Dependencies {
-  lazy val circeVersion = "0.8.0"
-
+  val scalaVersions = Seq("2.11.11", "2.12.4")
```
Comment (hosamaly) [inline project/Dependencies.scala]: We can now use `2.11.12` and `2.12.5`.

File: project/build.properties [hunk 5]
```diff
@@ -1 +1,2 @@
-sbt.version=0.13.16
+sbt.version=1.1.1
```
Comment (hosamaly) [inline project/build.properties]: [sbt 1.1.2](https://github.com/sbt/sbt/releases/tag/v1.1.2) was released today. :)

File: build.sbt [hunk 6]
```diff
@@ -27,20 +27,20 @@ lazy val utils = (project in file("utils"))
...
+      , Compile / unmanagedResourceDirectories       += (ThisBuild / baseDirectory).value / "build"
+      , Compile / unmanagedResources / excludeFilter := "*.fjs" || "*.scss"
       )
 
 lazy val coreSettings: Seq[Setting[_]] = 
   Seq ( name      := Metadata.ghProject
       , publishTo := sonatypePublishTo.value
       , WebKeys.pipeline := WebKeys.pipeline.dependsOn(webpack.toTask("")).value
-      , WebpackKeys.config in webpack := file("apps.config.js")
+      , WebpackKeys.config / webpack := file("apps.config.js")
```
Comment (hosamaly) [inline build.sbt]: I would've expected this to be `webpack / WebpackKeys.config`...

**Comment by tomrf1** [comment]
👍 looks sensible

**Pull request merged**